tests/data/test_publish.py

```diff
@@ -148,6 +148,53 @@ def extractor(self, mock_folder_raw_data: Path):
         return FolderExtractor(
             folder_raw_data=mock_folder_raw_data, pattern_raw_data="*.json"
         )
+    
+    @pytest.fixture
+    def fake_dataframe(self):
+        return pd.DataFrame([
+            {
+                "station_id": 1,
+                "num_bikes_available": 2,
+                "num_bikes_available_types": [
+                    {"mechanical": 1},
+                    {"ebike": 1},
+                ],
+            }
+        ])
+
+    def test_find_filepath(self, mocker: MockerFixture, extractor, mock_folder_raw_data):
+        mock_get_one_filepath = mocker.patch('velib_spot_predictor.data.publish.get_one_filepath')
+        mock_get_one_filepath.return_value = Path(f"{mock_folder_raw_data}/velib_availability_real_time_20220101-120001.json")
+        test_date = datetime(2022, 1, 1, 12, 0)
```

Comment:
You have 2 choices here:
1. mock get_one_filepath and not use the mock_folder_raw_data fixture because you will not interact anymore with the filesystem
2. Not mock get_one_filepath and you have to write data to the temporary folder so that it is detected by get_one_filepath.

My advices are:
- unit tests should have the least dependencies between each other and only test the behaviour of a single function
- unit tests should have the least possible interactions with the filesystem

Therefore, I think your mocking of get_one_filepath is preferable and the interaction (a fake one thanks to temporary directory) should be tested only for the get_one_filepath unit tests.

I agree that the tests that were already written for this precise class do not follow my own best practices. Feel free to change them too ;)

Continue this great work !